### PR TITLE
fix some issues reported by sonar

### PR DESCRIPTION
**Pull request by hasys**
Hi @romartin,

I fixed some issues found by Sonar: https://sonarcloud.io/dashboard?id=org.kie.workbench%3Akie-wb-common

This PR depends on https://github.com/kiegroup/kie-soup/pull/77

### CODE REVIEW

**Comment by romartin** [comment]
Adding a second reviewer - @tiagodolphine

**Comment by hasys** [comment]
Jenkins failed due to bug test fixed by https://github.com/kiegroup/kie-wb-common/pull/2331

Jenkins retest please

**Comment by romartin** [comment]
Build failed again, lemme know when it's ready @hasys , np! Thx

**Comment by hasys** [comment]
Random fail in Services project.

Jenkins retest please

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/canvas/controls/containment/CaseManagementContainmentAcceptorControlImpl.java [hunk 1]
```diff
@@ -241,8 +241,10 @@ int getAddIndex(final WiresShape wiresShape,
 
             // Otherwise, add to a stage vertically
             // And if it is the same stage
-            if (state.getOriginalParent().isPresent() && state.getOriginalParent().get().equals(container)) {
-                return parent.getOutEdges().size() - 1;
+            if (state.getOriginalParent().isPresent()) {
```
Comment (tiagodolphine) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/canvas/controls/containment/CaseManagementContainmentAcceptorControlImpl.java]: just  a suggestion on this block:

```suggestion
           return state.getOriginalParent()
                    .filter(container::equals)
                    .map(p -> parent.getOutEdges().size() - 1)
                    .orElse(parent.getOutEdges().size());
```

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/export/DelegateNativeContext2D.java [hunk 2]
```diff
@@ -80,13 +80,13 @@ public void initDeviceRatio() {
...
-        context.saveGroup(new HashMap<String, String>() {{
-            //setting the node id in case it exists on graph
-            Optional.ofNullable(canvasHandler.getGraphIndex().get(id)).ifPresent(node -> {
-                put(DEFAULT_NODE_ID, id);
-                put(svgNodeId, id);
-            });
-        }});
+        HashMap<String, String> map = new HashMap<>();
+        //setting the node id in case it exists on graph
+        Optional.ofNullable(canvasHandler.getGraphIndex().get(id)).ifPresent(node -> {
```
Comment (triceo) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/export/DelegateNativeContext2D.java]: In this case, `Optional` doesn't really give you anything. A simple null check does the exact same thing, doesn't allocate a new object on the heap, and doesn't require a lambda.

Also, this seems much better readable:
  
    String node = canvasHandler.getGraphIndex().get(id);
    if (canvasHandler.getGraphIndex().get(id) == null) {
         context.saveGroup(Collections.emptyMap());
    } else {
         HashMap<String, String> map = new HashMap<>(2);
         map.put(DEFAULT_NODE_ID, id);
         map.put(svgNodeId, id);
         context.saveGroup(map);
    }

Overall, this is also much more efficient that the original.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/export/DelegateNativeContext2D.java [hunk 3]
```diff
@@ -100,10 +100,10 @@ public void save() {
 
     public void save(String id) {
         context.saveStyle();
-        context.addAttributes(new HashMap<String, String>() {{
-            //setting the node id in case it exists on graph
-            Optional.ofNullable(id).ifPresent(node -> put(DEFAULT_NODE_ID, id));
-        }});
+        HashMap<String, String> map = new HashMap<>();
+        //setting the node id in case it exists on graph
+        Optional.ofNullable(id).ifPresent(node -> map.put(DEFAULT_NODE_ID, id));
```
Comment (triceo) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/export/DelegateNativeContext2D.java]: Dtto.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/wires/CaseManagementContainmentControl.java [hunk 4]
```diff
@@ -137,12 +137,14 @@ public Point2D getAdjust() {
 
     @Override
     public void execute() {
-        // Execute.
-        final CaseManagementShapeView ghost = state.getGhost().get();
-        //Children contains m_ghost and others excluding m_shape. This replaces m_ghost with m_shape.
-        final WiresContainer originalParent = state.getOriginalParent().get();
-        reparentDraggedShape(ghost,
-                             originalParent);
+        if (state.getGhost().isPresent() && state.getOriginalParent().isPresent()) {
```
Comment (tiagodolphine) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/wires/CaseManagementContainmentControl.java]: just a suggestion on this block:
```suggestion
state.getGhost()
    .ifPresent(ghost -> state.getOriginalParent()
         .ifPresent(originalParent -> reparentDraggedShape(ghost, originalParent)));
```

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/canvas/controls/containment/CaseManagementContainmentAcceptorControlImpl.java (hunk 1 above)
Comment (triceo) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/canvas/controls/containment/CaseManagementContainmentAcceptorControlImpl.java]: Alternatively:

    return state.getOriginalParent()
        .flatMap(original -> original.equals(container) ? Optional.of(parent.getOutEdges().size() - 1) : Optional.empty())
        .orElseGet(() -> parent.getOutEdges().size());

You get to have just one return and no nesting.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/wires/CaseManagementContainmentControl.java [hunk 5]
```diff
@@ -174,11 +176,13 @@ public void destroy() {
     }
 
     private void restore(final CaseManagementShapeView ghost) {
-        final WiresContainer originalParent = state.getOriginalParent().get();
-        final int originalIndex = state.getOriginalIndex().get();
-        restore(ghost,
-                originalParent,
-                originalIndex);
+        if (state.getOriginalParent().isPresent() && state.getOriginalIndex().isPresent()) {
```
Comment (tiagodolphine) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/wires/CaseManagementContainmentControl.java]: just a suggestion on this block:

```suggestion
state.getOriginalIndex()
    .ifPresent(originalIndex -> state.getOriginalParent()
        .ifPresent(originalParent -> restore(ghost, originalParent, originalIndex)));
```

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/canvas/controls/containment/CaseManagementContainmentAcceptorControlImpl.java (hunk 1 above)
Comment (triceo) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/canvas/controls/containment/CaseManagementContainmentAcceptorControlImpl.java]: Made a similar suggestion regarding to the same code. Yours is better. :-)

Comment (tiagodolphine) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/canvas/controls/containment/CaseManagementContainmentAcceptorControlImpl.java]: Hey @triceo , yeah we made the suggestions almost at the same time, just saw them now :)

**Comment by hasys** [comment]
Hi @triceo, @tiagodolphine,

thank you, good catches! I added changes as separate commit.

@romartin I also rebased changes from master since looks like previous build failed due to conflicts.

Thank you!

**Comment by hasys** [comment]
Hi @romartin,

just fixed conflicts (applied master version since it also fixed an issue in the file).

Jenkins execute full downstream build

**Comment by romartin** [comment]
please @triceo @tiagodolphine waiting for your feedback here, missing approvals if so.. thanks!

**Pull request merged**